Omitting a subgroup object necessitates reset (#1295)
moq-wg/moq-transport: draft-ietf-moq-transport.md

If a sender closes the stream before delivering all such objects to the QUIC
stream, it MUST use a RESET_STREAM or RESET_STREAM_AT
{{!I-D.draft-ietf-quic-reliable-stream-reset}} frame. This includes, but is
not limited to:

* An open Subgroup exceeding its Delivery Timeout
* Early termination of subscription due to an UNSUBSCRIBE message
* A publisher's decision to end the subscription early
* A SUBSCRIBE_UPDATE moving the subscription's End Group to a smaller Group or
  the Start Location to a larger Location

> * Omitting a Subgroup Object due to the subcriber's Forward State

Reviewer: If you the publisher opened a stream, presumably the forward state at that time was true, so does it make sense to say "A change in the Forward state from true to false"?

```suggestion
* A change in the Forward state from true to false
```

> * Omitting a Subgroup Object due to the subcriber's Forward State

Author: You made basically the same comment as Suhas so I have the same reply.  Simply changing the forward state is not what necessitates closing the stream. If an object is published every 1s, I can flip forward off and on again between objects and retain the stream.  The problem only happens when I go to publish an object on that subgroup and forward=0 --> then you have to reset.

Of course, your implementation can choose to reset upon receiving Forward=0 too, but nothing breaks if you leave it open longer.

> * Omitting a Subgroup Object due to the subcriber's Forward State

Reviewer 2: May be what I was intending to capture is 
If Forward state moves from 0 to 1 and there is a gap in the next object Id, then the stream needs to be reset

> * Omitting a Subgroup Object due to the subcriber's Forward State

Author: > If Forward state moves from 0 to 1 and there is a gap in the next object Id, then the stream needs to be reset

I tried to capture it with this:

> Omitting a Subgroup Object due to the subcriber's Forward State
